src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ComService.kt

```diff
@@ -153,31 +158,54 @@ class ComService(
     }
   }
 
-  private fun findPrisonersForUnstartedRecords(record: List<Pair<CaseloadResult, Licence?>>): Map<String, PrisonerSearchPrisoner> {
-    val prisonNumbers = record
-      .filter { (_, licence) -> licence == null }
-      .mapNotNull { (result, _) -> result.identifiers.noms }
-
-    val prisoners = this.prisonerSearchApiClient.searchPrisonersByNomisIds(prisonNumbers)
+  private fun findPrisonersWithoutDraftLicences(
+    eligiblePrisonersList: List<PrisonerSearchPrisoner>,
+    record: List<Pair<CaseloadResult, Licence?>>,
+  ): Map<String, PrisonerSearchPrisoner> {
+    val eligiblePrisoners =
+      eligiblePrisonersList.filter { prisoner ->
+        !record.isLicenceInGivenStatusPresent(prisoner.prisonerNumber, LicenceStatus.DRAFT_LICENCES) ||
```

Comment:
Aren't these checks the inverse of each other?

Reply:
yup, second condition is covered implicitly by first condition and probabation cases would be covered. so fixed it by removing second condition.